### DISCUSSION TITLE
Looking for assistance creating an app(s) using BudiBase.  WiIling to pay depending.

### DISCUSSION THREAD

**Discussion by PaulERayburn** [Budibase/budibase#8822] [General]
Hello, I have been attempting to create an app in BudiBase but have not been successful.  I have posted a few times in this discussion thread to try and get past some stumbling blocks https://github.com/Budibase/budibase/discussions/8734#discussion-4582489  If you give this a quick scan you should get a sense of what I am proposing and where I am at.
 I do not have any coding experience. I do have some contacts who can help a bit but I cannot rely on that.

I am trying to decide if BudiBase is the best tool for my needs.  I am willing to hire someone to A) help me decide if BB is practical by building a test project if need be.
B) Hire someone to assist me with building the final product.
I was hoping that the community could help a bit but I understand I have somewhat specific requirements and I have limited knowledge and time.
Can anyone make some recommendations? 

Here is a list of my basic requirements.

1) Scalability: Hopefully we are satisfied with BudiBase and use it to expand our basic app and add more features and more users. I think this is possible

2) Reliability: Must be a reliable app with easy backup and restore.

3)Secure: Security is important but for the basic app there is no highly confidential personal or financial information but the data cannot be openly shared with the public.  If we expand the app with new features this could get into some privacy issues so I would like to better understand that side.

4) Repeatability: Whatever we build I need to be able to document how it was created so that it can be re-created in the event of a major failure, maintained by either myself or any party with only basic coding skills.  Ie I should not be fully reliant upon a single person or party for system maintenance.

5) The basis functionality: Query Multiple Tables and refine the records to create a subset(s) with "parent-child" type relationships (my loose description). These relationships need to be either added to the source tables or added to a new table or both. These actions will be mostly manual eg. the User selects data, filters data and identifies matches then clicks to push that to create the relationships which are "permanent"(permanent means any subsequent searches of the source data tables will reveal those matches) those related/ matched records may be modified or added to subsequently.  The selected sets will also need to have the ability to export most likely to a CSV.  
Once these sets are identified they will also automatically create a cloud-based "folder" for each record where additional documents will be stored including images. 
The management of documents and images can be handled in an appropriate fashion based on recommendations for size, type, cost, and accessibility requirements.

At present, I have identified that an Amazon AWS RDS server with a Postgres database may be suitable and affordable.
We have created the server account and tied a few Postgres tables.  If you read my threads you'll have some idea how far I got on my own before realizing I was not going to get any further without just shooting in the dark.

Thanks in advance for any help or suggestions.

**Comment by PaulERayburn** [comment]
If I was looking to hire someone to assist with my BudiBase app implementation what skills would be best suited? I can imagine Javascript and Handlebars but if I could get some feedback that would be helpful.

**Comment by JayP718** [comment]
Hi Paul,

I've read your post but honestly it's a bit difficult to understand what you are trying to achieve. Here are my thoughts so far

2.  Budibase premium allows for automatic back ups, you might also benefit from backups of the actual database hosting the information.
  Reliability is pretty good from my experience using Budibase especially with how easy it is to create a back up of the app
3. From the aspect of security. That is possible using budibase cloud and if you are hosting it yourself you can restrict access by hosting it on your private network.
5. I'm having trouble really understanding what you are trying to do here but it seems doable using stored procedures and some type of data pipeline.
      -"Once these sets are identified they will also automatically create a cloud-based "folder" for each record where additional documents 
      will be stored including images."-Seems like you might have to do this using some sort of API but you might also be able to use the 
       bash automation within budibase

I think I probably have more questions than I answered :) but I don't mind to hop on a discord/zoom call to help out

**Reply by PaulERayburn** [reply]
Thanks for the reply. @JayP718  I was attempting to give a basic overview of what I am hoping to achieve as a base functional app.  So assuming the security and reliability are addressed the remaining issue is the functionality.  I have one project, described as #5 in my original post, that is basically key to moving forward with BudiBase and even if that was the only thing we achieved I would be happy with that.
What times and days are you available for a chat?   I am PST time zone but available for almost anytime.

**Reply by JayP718** [reply]
I've sent you a message via Discord. Let's continue on there

> Thanks for the reply. @JayP718  I was attempting to give a basic overview of what I am hoping to achieve as a base functional app.  So assuming the security and reliability are addressed the remaining issue is the functionality.  I have one project, described as #5 in my original post, that is basically key to moving forward with BudiBase and even if that was the only thing we achieved I would be happy with that.
> 
> What times and days are you available for a chat?   I am PST time zone but available for almost anytime.
> 
>
